zebra-network/src/policies.rs

```diff
@@ -34,9 +34,6 @@ impl<Req: Clone + std::fmt::Debug, Res, E: std::fmt::Debug> Policy<Req, Res, E>
                 Some(
                     // Let other tasks run, so we're more likely to choose a different peer,
                     // and so that any notfound inv entries win the race to the PeerSet.
-                    //
-                    // TODO: move syncer retries into the PeerSet,
-                    //       so we always choose different peers (#3235)
                     Box::pin(tokio::task::yield_now().map(move |()| retry_outcome)),
```

Comment:
This security fix was actually implemented, but in a different way:

```suggestion
                    // # Security
                    //
                    // We want to choose different peers for retries, so we have a better chance of getting each block.
                    // This is implemented by the connection state machine sending synthetic `notfound`s to the 
                    // `InventoryRegistry`, as well as forwarding actual `notfound`s from peers.
                    Box::pin(tokio::task::yield_now().map(move |()| retry_outcome)),
```